### PR TITLE
refactor: clean up portal-backend environment variables for vendor neutrality

### LINKED ISSUE #466: Clean up portal-backend environment variables for vendor neutrality
### Current Limitation

The `portal-backend` configuration surface carried Choreo-specific and stale environment variables that no longer matched how the service is actually run:

- A `.choreo/component.yaml` bound the deployment to WSO2 Choreo and injected DB/PDP/audit variables under Choreo-specific names (`CHOREO_OPENDIF_DB_*`, `CHOREO_PDP_CONNECTION_*`, `CHOREO_AUDIT_CONNECTION_SERVICEURL`) that **did not match the names the code reads** (`DB_*`, `PDP_SERVICEURL`) — so in deployment those values silently fell back to code defaults.
- `docker-compose.test.yml` used yet a third DB naming scheme (`CHOREO_DB_portal_backend_*`), also unmatched by the code, so the test container never connected to the intended DB.
- `.env.example` contained a dead `HOST` variable, a wrong `PORT` default (`8080` vs the actual `3000`), and "Asgardeo"-labelled comments left over before the vendor-neutrality rename.
- `main.go` referenced a non-existent `ENABLE_AUDIT` flag in a comment; the `Dockerfile` set unused `CONFIG_DIR`/`ENVIRONMENT` vars.
- The PDP client required a `PDP_API_KEY` (`apikey` header), which is unnecessary now that the PDP is reached through a trusted API gateway.

### Suggested Improvement

Normalize the environment variables to vendor-neutral names and remove dead/unnecessary config (implemented in the current changes):

- **Deleted** `.choreo/` folder.
- **Renamed** to vendor-neutral, consistent names (`SERVICEURL` → `SERVICE_URL`):
  - `CHOREO_PDP_CONNECTION_CHOREOAPIKEY` → `PDP_API_KEY` (then removed, see below)
  - `PDP_SERVICEURL` → `PDP_SERVICE_URL`
  - `CHOREO_AUDIT_CONNECTION_SERVICEURL` → `AUDIT_SERVICE_URL`
- **Fixed `docker-compose.test.yml`** to use the DB names the code reads (`DB_HOST/PORT/USERNAME/PASSWORD/NAME`).
- **Cleaned `.env.example`**: removed dead `HOST`, set `PORT=3000`, renamed "Asgardeo" comments to "IDP".
- **Removed dead config**: `ENABLE_AUDIT` comment in `main.go`; `CONFIG_DIR`/`ENVIRONMENT` in the `Dockerfile`.
- **Removed `PDP_API_KEY` entirely** (env var, `PDPService.apiKey` field, and the `apikey` auth header) — the PDP now sits behind a trusted API gateway.
- Updated `README.md` accordingly and adjusted all affected tests. Build, vet, and tests pass.

> **Note (breaking):** Deployments must update their environment variable names — `PDP_SERVICE_URL` and `AUDIT_SERVICE_URL` replace the old Choreo names, and `PDP_API_KEY` / `PDP_SERVICEURL` are no longer read.

### Version

Based on commit `45216139` (branch `main`).

### Additional Context

Follows the earlier vendor-neutrality work (#463, #465, #460). Files touched: `.choreo/component.yaml` (deleted), `.env.example`, `Dockerfile`, `README.md`, `docker-compose.test.yml`, `main.go`, `v1/handlers/v1_handler.go`, `v1/services/pdp_service.go`, and related tests.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Normalizes the `portal-backend` configuration surface: removes Choreo-specific and stale environment variables, renames the remaining ones to vendor-neutral, consistent names, and drops the PDP API key now that the PDP is reached through a trusted API gateway. This resolves several mismatches where the names injected at deployment did not match the names the code actually reads (causing silent fallback to defaults).

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- **Removed** `.choreo/component.yaml` (drops the WSO2 Choreo binding).
- **Renamed** environment variables to vendor-neutral names, standardizing `SERVICEURL` → `SERVICE_URL`:
  - `PDP_SERVICEURL` → `PDP_SERVICE_URL`
  - `CHOREO_AUDIT_CONNECTION_SERVICEURL` → `AUDIT_SERVICE_URL`
- **Removed `PDP_API_KEY` entirely** — the env var, the `PDPService.apiKey` field, and the `apikey` auth header. The PDP now sits behind a trusted API gateway. (This retires the former `CHOREO_PDP_CONNECTION_CHOREOAPIKEY`.)
- **Fixed `docker-compose.test.yml`** to use the DB variable names the code reads (`DB_HOST/PORT/USERNAME/PASSWORD/NAME`) instead of the old `CHOREO_DB_portal_backend_*` names.
- **Cleaned `.env.example`**: removed dead `HOST`, corrected `PORT` default to `3000`, renamed "Asgardeo" comments to "IDP".
- **Removed dead config**: the non-existent `ENABLE_AUDIT` reference in `main.go`; unused `CONFIG_DIR`/`ENVIRONMENT` in the `Dockerfile`.
- **Updated `README.md`** and adjusted all affected tests (`NewPDPService` is now single-arg; obsolete missing-key test removed).

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

`go build ./...`, `go vet ./...`, and `go test ./v1/...` all pass.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #466

## Additional Notes

No functional behavior changes beyond dropping the PDP API key auth header; the rest is renaming and dead-config removal.

## Deployment Notes

**Breaking for deployments** — environment variable names must be updated:

- `PDP_SERVICEURL` → `PDP_SERVICE_URL`
- `CHOREO_AUDIT_CONNECTION_SERVICEURL` → `AUDIT_SERVICE_URL`
- `PDP_API_KEY` / `CHOREO_PDP_CONNECTION_CHOREOAPIKEY` are **no longer read** and should be removed.
- DB values must be provided via `DB_HOST/DB_PORT/DB_USERNAME/DB_PASSWORD/DB_NAME` (no longer the `CHOREO_*` names).
- Since `.choreo/component.yaml` is removed, Choreo deployments now rely on externally configured environment variables using the names above.

### CODE REVIEW

**Review by ginaxu1 — DISMISSED** [review]

File: portal-backend/v1/handlers/v1_handler.go [hunk 1]
```diff
@@ -100,17 +100,12 @@ func NewV1Handler(db *gorm.DB) (*V1Handler, error) {
 	}
 	memberService := services.NewMemberService(db, idpProvider)
 
-	pdpServiceURL := os.Getenv("PDP_SERVICEURL")
+	pdpServiceURL := os.Getenv("PDP_SERVICE_URL")
 	if pdpServiceURL == "" {
-		return nil, fmt.Errorf("PDP_SERVICEURL environment variable not set")
+		return nil, fmt.Errorf("PDP_SERVICE_URL environment variable not set")
 	}
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portal-backend/v1/handlers/v1_handler.go:106]: Addressed

File: portal-backend/v1/services/pdp_service.go [hunk 2]
```diff
@@ -17,26 +17,19 @@ import (
...
-func NewPDPService(baseURL string, apiKey string) *PDPService {
+// NewPDPService creates a new instance of PDPService.
+// The PDP is reached through a trusted API gateway, so no API key is required.
+func NewPDPService(baseURL string) *PDPService {
 	return &PDPService{
 		baseURL:    baseURL,
-		apiKey:     apiKey,
 		HTTPClient: &http.Client{Timeout: 10 * time.Second},
 	}
 }
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portal-backend/v1/services/pdp_service.go:34]: Addressed

**Review by ginaxu1 — APPROVED** [review]
